development/development/phpstorm.rst

```diff
@@ -0,0 +1,212 @@
+
+5. Select the phpBB project folder you cloned from GitHub and click "Project Root". Then click "Finish".
+
+Code Style
+==========
+
+Setting up PhpStorm to honor phpBB's coding style guide will ensure you are always writing code that meets phpBB's requirements. These settings are configured in phpStorm's Options/Preferences, under **Editor > Code Style**.
+
+.. note:: Before changing settings, it's a good idea to create a phpBB "Scheme" in the Code Style settings pane. This phpBB scheme can be applied to any phpBB project you create in phpStorm.
+
+.. seealso:: For your convenience we have provided an XML export of the following code style settings for phpBB (see`phpBB Code Style Scheme`_). You can import these settings into your project and all the following styling settings will be configured for you.
+
+PHP
+###
+
+Tabs & Indents:
+***************
+
+Tabs should be used (not spaces). All tab and indent sizes should be set to 4 spaces. Also be sure "Keep indents on empty lines" is NOT checked.
+
+Spaces:
+*******
+
+There are many settings for handling spaces, and the defaults should work for the most part. The general rule is single-spaces should surround all operators and parenthesis (except for function declarations/calls).
+
+Wrapping and Braces:
+********************
+
+The general rule here is that braces always go on new lines.
+
+CSS, JAVASCRIPT, HTML, TWIG
+###########################
+
+Tabs & Indents:
+***************
+
+Tabs should be used (not spaces). All tab and indent sizes should be set to 4 spaces. Also be sure "Keep indents on empty lines" is NOT checked.
+
+JSON, YAML
+##########
+
+Json and Yaml files uses 4 spaces over tabs, so tab character should be disabled and 4 spaces the indent size.
+
+Inspections
+===========
+
+One of the most powerful features of an IDE like PhpStorm is its ability to inspect your code and provide, in real time, errors and warnings. An IDE is fully aware of how phpBB's codebase is designed to work, and how all of its classes and functions interact with each other. Inspections can instantly show you, for example, if you are using a function incorrectly, calling non-existent methods, missing arguments, etc.
+
+The default inspection settings should work just fine. However there are a couple adjustments that might be desired:
+
+* If phpStorm is not connected to your database server, you may want to turn off **SQL > SQL Dialect Detection** and **SQL > No data sources configured warnings**.
+* phpBB uses fully qualified namespaces, so you can turn off this inspection warning **PHP > Code Style > Unnecessary fully qualified name**.
+* You may enable additional JavaScript inspections. Under **JavaScript > Code quality tools**, you can enable JSCS, JSHint, JSlint and/or ESLint.
```

Comment:
We actually recommend JSHint and JSCS as we provide predefined inspection settings for these.